util/tele/composite_logger.go

```diff
@@ -20,70 +20,70 @@ import (
 	"github.com/go-logr/logr"
 )
 
-type compositeLogger struct {
-	loggers []logr.Logger
+type compositeLogSink struct {
+	logSinks []logr.LogSink
 }
 
-func (c *compositeLogger) Enabled() bool {
-	for _, l := range c.loggers {
-		if !l.Enabled() {
+func (c *compositeLogSink) Init(info logr.RuntimeInfo) {
+	info.CallDepth = info.CallDepth + 2
+	for _, l := range c.logSinks {
+		l.Init(info)
+	}
+}
+
+func (c *compositeLogSink) Enabled(v int) bool {
+	for _, l := range c.logSinks {
+		if !l.Enabled(v) {
 			return false
 		}
 	}
 	return true
 }
 
-func (c *compositeLogger) iter(fn func(l logr.Logger)) {
-	for _, l := range c.loggers {
-		// the callDepthLogger interface allows us to change the depth of the stack so that we can get the real
-		// line where the log statement was called. We need to do this because the composite logger adds to the
-		// call stack due to wrapping the internal logger.
-		l = logr.WithCallDepth(l, 3)
+func (c *compositeLogSink) iter(fn func(l logr.LogSink)) {
+	for _, l := range c.logSinks {
 		fn(l)
 	}
 }
 
-func (c *compositeLogger) Info(msg string, keysAndValues ...interface{}) {
-	c.iter(func(l logr.Logger) {
-		l.Info(msg, keysAndValues...)
+func (c *compositeLogSink) Info(level int, msg string, keysAndValues ...interface{}) {
```

Comment:
is this meant to replace [log.V(n).Info(msg)](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e123e78fcee9a9bb214999deb8c35fc6f1bffba4/azure/services/agentpools/agentpools.go#L162) with `log.Info(n,msg)`? Should we do a followup for such replacements?

Reply:
No. The Logger struct [still has a `V(n)` function](https://github.com/go-logr/logr/blob/master/logr.go#L277). It's only the LogSink interface that has [Info(n, msg)](https://github.com/go-logr/logr/blob/master/logr.go#L421).

For an example of how this works, this the example logger implementation [here](https://github.com/go-logr/logr/blob/master/examples/tab_logger.go#L46) and example usage [here](https://github.com/go-logr/logr/blob/master/examples/usage_example.go#L159) and [here](https://github.com/go-logr/logr/blob/master/examples/usage_example.go#L76).